build-common.xml

```diff
@@ -1214,12 +1214,27 @@ information.
 			</if>
 
 			<local name="junit.jvm.bin" />
+			<local name="junit.java.home" />
 
-			<condition else="${junit.jvm}" property="junit.jvm.bin" value="${junit.jvm.override}">
-				<isset property="junit.jvm.override" />
-			</condition>
+			<if>
+				<equals arg1="${junit.jvm.override}" arg2="" />
+				<then>
+					<property name="junit.jvm.bin" value="" />
+					<property name="junit.java.home" value="${env.JAVA_HOME}" />
+				</then>
+				<else>
+					<property name="junit.jvm.bin" value="${junit.jvm.override}" />
+					<propertyregex
+						input="${junit.jvm.override}"
+						property="junit.java.home"
+						regexp="(.*)[\\/]bin[\\/]java"
+						select="\1"
```

Comment:
junit.jvm.override will be for example: `/opt/java/java21-zulu/bin/java`
We need the java home path without `/bin/java` to pass into JAVA_HOME env variable: `/opt/java/java21-zulu`

[\\/] matches both `\` and `/`